ODIN_II/SRC/output_blif.cpp

```diff
@@ -65,25 +65,25 @@ static bool warn_undriven(nnode_t* node, nnet_t* net) {
     return false;
 }
 
-// TODO Uncomment this for In Outs
-//static void merge_with_inputs(nnode_t* node, long pin_idx) {
-//    oassert(pin_idx < node->num_input_pins);
-//    nnet_t* net = node->input_pins[pin_idx]->net;
-//    warn_undriven(node, net);
-//    // Merge node with all inputs with fanout of 1
-//    if (net->num_fanout_pins <= 1) {
-//        for (int i = 0; i < net->num_driver_pins; i++) {
-//            npin_t* driver = net->driver_pins[i];
-//            if (driver->name != NULL && ((driver->node->type == MULTIPLY) || (driver->node->type == HARD_IP) || (driver->node->type == MEMORY) || (driver->node->type == ADD) || (driver->node->type == MINUS))) {
-//                vtr::free(driver->name);
-//                driver->name = vtr::strdup(node->name);
-//            } else {
-//                vtr::free(driver->node->name);
-//                driver->node->name = vtr::strdup(node->name);
-//            }
-//        }
-//    }
-//}
+static void merge_with_inputs(nnode_t* node, long pin_idx) {
+    oassert(pin_idx < node->num_input_pins);
+    nnet_t* net = node->input_pins[pin_idx]->net;
+    warn_undriven(node, net);
+    // Merge node with all inputs with fanout of 1
+    if (net->num_fanout_pins <= 1) {
+        for (int i = 0; i < net->num_driver_pins; i++) {
+            npin_t* driver = net->driver_pins[i];
+            if (driver->name != NULL && ((driver->node->type == MULTIPLY) || (driver->node->type == HARD_IP) || (driver->node->type == MEMORY) || (driver->node->type == ADD) || (driver->node->type == MINUS))) {
```

Comment:
given you are calling it on the output node only, this will never be reached.
I also don't forsee this being used for anything else but output. can you assert that node->type is OUTPUT_NODE
I would rather not see this used anywhere else in the future unless theres a good reason for it, so guarding it should make that clear

Reply:
I assume you mean the input node? The driver can be anything that drives an output yeah?